src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -660,17 +894,17 @@ avtUnstructuredDomainBoundaries::ExchangeScalar(vector<int>         domainNum,
     switch (maxDataType)
     {
         case VTK_INT:
-            return ExchangeData_int(domainNum, isPointData, scalars);
+            return ExchangeData<int>(domainNum, isPointData, scalars);
```

Comment:
I am worried a bit about the limited set of type cases this logic covers here. I did a quick inventory of our database plugins and below are types our plugins can instantiate that we do not appear to cover here. Is this a problem?

```
vtkBitArray
vtkIdTypeArray
vtkLongArray
vtkLongLongArray
vtkShortArray
vtkStringArray
vtkUnsignedLongLongArray
vtkUnsignedShortArray
```

Reply:
If any of those are used as arrays for field values, then we could be in trouble. Unless there is a conversion happening somewhere for arrays that don't have the chosen types. I don't think we've run into issues with incompatible types before; I've never seen (or had a user report seeing) that exception.

Reply:
I am pretty sure we use `vtkBitArray` for enumerated scalars (which turn out to be a scalar mesh variable). I think `LongLong` may be used for global ids in several plugins too. Those also often come along as scalar mesh variables. 

Now, do we encounter those things *AND* parallel decomposed UCD mesh data that *DOES NOT ALSO* have its own ghost zone data already? I suspect that is rare. But, if we ever do encounter it, I think its a problem here. And, I think it is probably sufficient to add cases for most of those types...`StringArray` and `BitArray` probably require more thought. Maybe add the easily add-able cases and file an issue ticket for the more thought cases? I dunno, that ticket will probably disappear into the noise too. We should talk about at an upcomming meeting I think.

Reply:
I made a note to bring this up at our next project meeting.